packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -141,49 +161,81 @@ async def unarchive_dir(
                 for zip_entry in zip_file_handler.infolist()
             ]
 
-            extracted_paths: Tuple[Path] = await asyncio.gather(*tasks)
+            extracted_paths: list[Path] = await tqdm_asyncio.gather(
```

Comment:
this tqdm is really elegant! the ``tenacity`` of progress monitoring :-)

Reply:
definitely!